src/Provider/GoogleMaps/GoogleMaps.php

```diff
@@ -257,6 +257,9 @@ private function fetchUrl($url, $locale, $limit, $region)
             if (isset($result->formatted_address)) {
                 $address = $address->withFormattedAddress($result->formatted_address);
             }
+            if (property_exists($builder, 'subpremise')) {
```

Comment:
I see, this is quite nice. But property overloading like this is quite slow. 

Lets add a new property to the `AddressBuilder` that can take arbitrary data. We also need `AddressBuilder::setValue($name, $value)`, `AddressBuilder::hasValue($name)`, `AddressBuilder::getValue($name, $default=null)`.